### ISSUE TITLE
Try latency compensation on marking messages as read

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#685]
As described in #81, when we implement marking messages as read when you scroll past them, we'll have the unread markers disappear optimistically without waiting for an ack from the server:

> * Unlike most everything else in the Zulip data model, apply "latency compensation" when marking messages as read on scroll, meaning eagerly apply the change to the local UI rather than wait for the server to acknowledge it with an event.
>   * Specifically, eagerly fade out the unread-message markers in the message list.

This issue is for going further, to eagerly update the model itself.

* The zulip-mobile RN app doesn't do this; the consequence is that it looks kind of glitchy as the "N unreads" banner at the top of the message list doesn't update or disappear in sync with the unread markers.
* I'm not actually sure what the web app does.
* The downside is that if we eagerly update the model, then it's no longer part of a coherent snapshot together with the rest of our model — we'd be giving up some of the consistency provided by the Zulip event system. 

There are a couple of approaches we could take to that consistency question, but we should think through the consequences.

When doing this, we should also do the same with the "mark as read" button at the bottom of the message list. As part of that, we'll make its fade-out animation somewhat longer again. See #613.

**Comment by chrisbobbe** [comment]
[#mobile-team > last code changes for launch @ 💬](https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/last.20code.20changes.20for.20launch/near/2193892)
> There's one other gap in that implementation #F1574 relative to our original #F81 plan (and the legacy app), which is that it doesn't eagerly mark the messages as read locally, and instead waits for the event from the server.
> 
> In manual testing, the UX that way seems basically fine when you're on a good network like here in the office.
> 
> So I think that will be fine for launch. We'll file an M5b issue for adding that latency compensation.
> 
> (Adding it is potentially nontrivial without compromising performance, because of how our unreads data structure currently works. Totally solvable problem, but makes it a bigger task than I'd like to include between now and Monday.)

**Comment by gnprice** [comment]
The status quo after #1574 (for #81) will go a bit less far than the original plan: unlike the legacy app, it won't do latency compensation on the unread-message markers within the message list, either. So from a user-impact perspective, this issue is for causing those to update eagerly, as well as everything else in the UI (like the "Mark as read" button at the bottom of the message list, and the inbox page, and so on).

The content of how to implement this issue remains the same as originally planned above: eagerly update the app's local model to mark the messages as read. That will naturally have the consequence of causing those markers to eagerly update, along with the rest of the UI.
